src/core/pointcloud/qgslazdecoder.cpp

```diff
@@ -397,7 +399,7 @@ void decodePoint( char *buf, int lasPointFormat, char *dataBuffer, std::size_t &
         lazStoreToStream_<unsigned char>( dataBuffer, outputOffset, requestedAttribute.type, isLas14 ? p14.eofFlag() : p10.edge_of_flight_line );
         break;
       case QgsLazDecoder::LazAttribute::ScanAngleRank:
-        lazStoreToStream_<char>( dataBuffer, outputOffset, requestedAttribute.type, char( isLas14 ? p14.scanAngle() : p10.scan_angle_rank ) );
+        lazStoreToStream_<float>( dataBuffer, outputOffset, requestedAttribute.type, isLas14 ? p14.scanAngle() * 0.006f : p10.scan_angle_rank );
```

Comment:
Can you add a comment like `according to laz spec, scan angle is stored in 0.006 degrees increments` to justify the magic number?

Reply:
Added.